### DISCUSSION TITLE
Reworked documentation

### DISCUSSION THREAD

**Discussion by reichhartd** [orval-labs/orval#529] [Ideas]
[A few days ago Docusaurus v2 was released.](https://docusaurus.io/blog/2022/08/01/announcing-docusaurus-2.0)
I find this a good occasion to migrate and extend the website to Docusaurus.

Advantages of this would be:

-   Less maintenance effort. Many components are provided by Docusaurus out of the box, and some of the current custom components would not be needed anymore.
-   The current design is very similar to the default Docusaurus template, so the design changes would not be that large.
-   The documentation would be easier to understand with new components like [tabs](https://docusaurus.io/docs/next/markdown-features/tabs).
-   There are plugins like the [Algolia search](https://docusaurus.io/docs/search#using-algolia-docsearch) or an integration for [Google Analytics](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-google-analytics).
-   By making the documentation simpler, it would invite new contributors to customize or extend it.

Also I would extend the documentation with some markdown pages, or at least placeholders for follow up pull requests, what I have learned so far about Orval from my own experience or from the examples, like https://github.com/anymaniax/orval/issues/515.

The new documentation could then be released with the next major version.

**Comment by anymaniax** [comment]
Hey @reichhartd would be super cool to have a new version for the website. You would try to do it? Or you want me to take a look to it?

**Reply by reichhartd** [reply]
If you want to, you are welcome to take over the task.

**Reply by anymaniax** [reply]
it will really depend on the time that I can assign to it
